resources/WindowsService/Program.cs

```diff
@@ -0,0 +1,263 @@
+﻿using System.ServiceProcess;
+using System.Text.Json;
+using System.Text.Json.Nodes;
+using System.Text.Json.Schema;
+using System.Text.Json.Serialization;
+using System.Text.Json.Serialization.Metadata;
+
+const int EXIT_OK = 0;
+const int EXIT_INVALID_ARG = 1;
+const int EXIT_INVALID_INPUT = 2;
+
+if (args.Length < 1)
+{
+    WriteError("Usage: WindowsService.exe <action>");
+    Environment.Exit(EXIT_INVALID_ARG);
+}
+
+var operation = args[0].ToLowerInvariant();
+WindowsServices? windowsServices = null;
+WriteDebug( $"Operation: {operation}");
+
+switch (operation)
+{
+    case "get":
+        if (args.Length < 3 || args[1].ToLowerInvariant() != "-input")
+        {
+            WriteError("Usage: WindowsService.exe get -input <jsonInput>");
+            Environment.Exit(EXIT_INVALID_ARG);
+        }
+        var jsonInput = args[2];
+        WriteTrace($"Input JSON: {jsonInput}");
+        if (!string.IsNullOrEmpty(jsonInput))
+        {
+            try
+            {
+                windowsServices = JsonSerializer.Deserialize<WindowsServices>(jsonInput);
+            }
+            catch (JsonException ex)
+            {
+                WriteError($"Failed to deserialize JSON: {ex.Message}");
+                Environment.Exit(EXIT_INVALID_ARG);
+            }
+        }
+
+        WindowsServices resultServices = new WindowsServices();
+        foreach (var service in windowsServices?.Services ?? Enumerable.Empty<WindowsService>())
+        {
+            if (service.Name is null)
+            {
+                WriteError("Property 'name' is required for 'get' operation.");
+                Environment.Exit(EXIT_INVALID_ARG);
+            }
+
+            var resultService = GetService(service.Name);
+            if (resultService is not null)
+            {
+                resultServices.Services.Add(resultService);
+            }
+        }
+
+        var options = new JsonSerializerOptions
+        {
+            DefaultIgnoreCondition = JsonIgnoreCondition.WhenWritingNull,
+        };
+        string json = JsonSerializer.Serialize(resultServices, options);
+        Console.WriteLine(json);
+
+        break;
+    case "schema":
```

Comment:
Sugget adding a newline between these for readability 

```suggestion
        break;
        
    case "export":
```